### DISCUSSION TITLE
Language pages with country code - there exists no portal for the URL

### DISCUSSION THREAD

**Discussion by sobocinski** [sulu/sulu#5695] [Q&A] [answered]
Hi, 
I have a question about languages, I'm trying to implement Sulu using "language" and "country" codes. 
For example domain.com/en-us. Everything works fine in the admin panel, but I have two problems:
* default language detection for `domain.com` does't work, message error is `There exists no portal for the URL "/"`, running `domain.com/en-us/` works fine
* Symfony profiler does't work localhost/_profiler - "There exists no portal for the URL"/_profiler" - it is possible that this issue is related to the one above?

My webspace xml looks like this:

```xml
<?xml version="1.0" encoding="utf-8"?>
<webspace xmlns="http://schemas.sulu.io/webspace/webspace"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://schemas.sulu.io/webspace/webspace http://schemas.sulu.io/webspace/webspace-1.1.xsd">
   
    <name>domain.com</name>
    <key>domain</key>

    <localizations>
        <localization language="en" country="us" default="true"/>
        <localization language="en" country="ca"/>
        <localization language="es" country="us"/>
        <localization language="pt" country="pt"/>
        <localization language="pt" country="br"/>
        <localization language="pl" country="pl"/>
    </localizations>

    ...

    <portals>
        <portal>
            <name>domain.com</name>
            <key>domain</key>
            <environments>
                <environment type="prod">
                    <urls>
                        <url>{host}/{language}-{country}</url>
                    </urls>
                </environment>
                <environment type="dev">
                    <urls>
                        <url>{host}/{language}-{country}</url>
                    </urls>
                </environment>
            </environments>
        </portal>
    </portals>

</webspace>
```
Do I have to specify the default language in <portal>, have someone had such a problem before?

**Comment by alexander-schranz** [comment, marked as answer]
Congratulation you are the first one created a discussion 🎉 

The `There exists no portal for the URL "/"` means that for the `/_profiler` url could not any webspace found when the `requestAnalyzer` validates the request in [sulu routerlistener](https://github.com/sulu/sulu/blob/7c24e0bf5ec47d13149348ecbabe69eb7ee5cca0/src/Sulu/Bundle/WebsiteBundle/EventListener/RouterListener.php#L58-L60), which is called for any request currently, maybe we need to discuss to disable this for specific routes. As this seems currently a problem with your `webspace` configuration as `{language}-{country}` is not Partial Match of a webspace as it should so I think in your case the simpliest way would be to switch to which have the same urls generated as you would expect:

```xml
                        <url>{host}/{localization}</url>
```

Another way when running into this issue for static routes, would be disable the [`_requestAnalyzer`](https://github.com/sulu/sulu/blob/85bb05f28ba04eb2f671b94bf8a359030e629545/src/Sulu/Bundle/MediaBundle/Resources/config/routing_website.yml#L13) for the [_profiler](https://github.com/sulu/skeleton/blob/66b1916e504292b909987891ef167561e5fbb7ce/config/routes/dev/web_profiler.yaml#L5-L7) url:

```yaml
web_profiler_profiler:
    resource: '@WebProfilerBundle/Resources/config/routing/profiler.xml'
    prefix: /_profiler
    defaults:
        _requestAnalyzer: false
```

not sure if this is possible for the profiler but for other.

**Reply by sobocinski** [reply]
@alexander-schranz looks like `<url>{host}/{localization}</url>` insted of `<url>{host}/{language}-{country}</url>` solved the problem. (
It was quick, thank you.

**Reply by alexander-schranz** [reply]
Good to hear that it helps I updated the answer a little bit to make it more clear where things happens and will mark it as accepted answer.
